### PR TITLE
test(query/stdlib/influxdata/influxdb): update rules_test to use algo-w

### PR DESCRIPTION
**Pull request by wolffcm**
This updates our rules test to create function expressions using the Rust parser and analysis, so that we are testing our rules against the kinds of things we'll actually see in production.

### CODE REVIEW

File: query/stdlib/influxdata/influxdb/rules_test.go [hunk 1]
```diff
@@ -657,7 +491,6 @@ func TestPushDownFilterRule(t *testing.T) {
 	for _, tc := range tests {
 		tc := tc
 		t.Run(tc.Name, func(t *testing.T) {
-			t.Parallel()
```
Comment (jlapacik) [inline query/stdlib/influxdata/influxdb/rules_test.go]: Making tests run serially isn't critical to tests passing right?

Comment (wolffcm) [reply query/stdlib/influxdata/influxdb/rules_test.go]: I think what was happening was that we were re-using function expressions in different cases, and they were getting transformed in-place. Unless we wrap those exprs in closures, the test will have issues.

Comment (wolffcm) [reply query/stdlib/influxdata/influxdb/rules_test.go]: I guess what I'm saying is that I'm okay with not letting them run in parallel, and leaving the test as-is.

Comment (jlapacik) [reply query/stdlib/influxdata/influxdb/rules_test.go]: That works for me.

**Pull request merged**